cmd/juju/application/deploy_test.go

```diff
@@ -606,6 +607,20 @@ func setupConfigFile(c *gc.C, dir string) string {
 	return path
 }
 
+type v1ToV2Locator struct {
```

Comment:
For this case, we can also use bakery.v2-unstable but we'll need to tweak charmrepo csclient to use a v2 bakery client. In that case, we'll rev charmrepo to v3, and import that into juju as a dep instead of v2